### PR TITLE
Clean up code action service and rationalize how we disallow certain Roslyn code actions

### PR DESCRIPTION
**Pull request by DustinCampbell**
Fixes https://github.com/OmniSharp/omnisharp-vscode/issues/925

There's a fair amount of refactoring in this PR but most of it is moving code around.

### CODE REVIEW

File: src/OmniSharp.Abstractions/Extensions/LoggerExceptions.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace Microsoft.Extensions.Logging
+{
+    public static class LoggerExceptions
+    {
+        public static void LogError(this ILogger logger, Exception ex, string message, params object[] args)
```
Comment (david-driscoll) [inline src/OmniSharp.Abstractions/Extensions/LoggerExceptions.cs]: I hate that this overload doesn't exist.

File: src/OmniSharp.Roslyn.CSharp/CodeActions/CodeActionHelper.cs [hunk 2]
```diff
@@ -1,102 +1,32 @@
...
             "Microsoft.CodeAnalysis.CSharp.CodeFixes.Async.CSharpConvertToAsyncMethodCodeFixProvider",
             "Microsoft.CodeAnalysis.CSharp.CodeFixes.Iterator.CSharpChangeToIEnumerableCodeFixProvider",
-            "Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ChangeSignature.ChangeSignatureCodeRefactoringProvider",
+            "Microsoft.CodeAnalysis.ChangeSignature.ChangeSignatureCodeRefactoringProvider",
+            "Microsoft.CodeAnalysis.ExtractInterface.ExtractInterfaceCodeRefactoringProvider"
+        };
+
+        private static readonly HashSet<string> _nrefactoryListToRemove = new HashSet<string>
+        {
             "ICSharpCode.NRefactory6.CSharp.Refactoring.CreateMethodDeclarationAction",
```
Comment (david-driscoll) [inline src/OmniSharp.Roslyn.CSharp/CodeActions/CodeActionHelper.cs]: Can probably just drop this list for now (perhaps even the hashset).

Currently Refactoring Essentials isn't built with the latest roslyn dlls, so the plugins branch fails.  I'll have to get some other form of unit test in there to ensure that plugins are working.

File: src/OmniSharp.Abstractions/Extensions/LoggerExceptions.cs (hunk 1 above)
Comment (filipw) [inline src/OmniSharp.Abstractions/Extensions/LoggerExceptions.cs]: i think the intent was to use const "categories" as event IDs, but yeah

Comment (DustinCampbell) [inline src/OmniSharp.Abstractions/Extensions/LoggerExceptions.cs]: Yeah, I hate the event ID thing. Should be optional. There's no reason to have a "0" everywhere if you aren't using them. :smile:

File: src/OmniSharp.Roslyn.CSharp/CodeActions/CodeActionHelper.cs (hunk 2 above)
Comment (DustinCampbell) [inline src/OmniSharp.Roslyn.CSharp/CodeActions/CodeActionHelper.cs]: Done.

**Pull request merged**